src/components/WalletAuthModal.tsx

```diff
@@ -38,15 +34,70 @@ export const WalletAuthModal = ({ open }: { open: boolean }) => {
   const classes = useStyles();
   const [connectMessage, setConnectMessage] = useState<string>('');
 
+  const [defaultChain, setDefaultChain] = useState<string>('1');
+
   const { showError, showInfo } = useApeSnackbar();
   const web3Context = useWeb3React<Web3Provider>();
   const walletAuth = useWalletAuth();
 
   const [isMetamaskEnabled, setIsMetamaskEnabled] = useState<boolean>(false);
 
+  const isMultichainEnabled = isFeatureEnabled('multichain_login');
+
+  const UNSUPPORTED = 'unsupported';
+  const UnsupportedNetwork = defaultChain == UNSUPPORTED;
+  const supportedChains = Object.entries(loginSupportedChainIds).map(key => {
+    return { value: key[0], label: key[1], disabled: false };
+  });
+
+  const loginOptions = concat(supportedChains, [
+    { value: UNSUPPORTED, label: '-', disabled: true },
+  ]);
+
+  const getInitialChain = async (ethereum: any) => {
+    if (ethereum) {
+      updateChain(await new Web3Provider(ethereum).send('eth_chainId', []));
```

Comment:
you could also do [`.getNetwork()`](https://docs.ethers.io/v5/single-page/#/v5/api/providers/-%23-providers--networks) and then converting from hex becomes unnecessary

Reply:
nice. will use